window/src/os/wayland/copy_and_paste.rs

```diff
@@ -102,9 +102,12 @@ impl CopyAndPaste {
 impl WaylandState {
     pub(super) fn resolve_copy_and_paste(&mut self) -> Option<Arc<Mutex<CopyAndPaste>>> {
         let active_surface_id = self.active_surface_id.borrow();
-        let active_surface_id = active_surface_id.as_ref().unwrap();
-        if let Some(pending) = self.surface_to_pending.get(&active_surface_id) {
-            Some(Arc::clone(&pending.lock().unwrap().copy_and_paste))
+        if let Some(active_surface_id) = active_surface_id.as_ref() {
+            if let Some(pending) = self.surface_to_pending.get(&active_surface_id) {
+                Some(Arc::clone(&pending.lock().unwrap().copy_and_paste))
+            } else {
+                None
+            }
```

Comment:
let's simplify this:

```suggestion
        let active_surface_id = self.active_surface_id.borrow();
        let active_surface_id = active_surface_id.as_ref()?;
        let pending = self.surface_to_pending.get(&active_surface_id)?;
        Some(Arc::clone(&pending.lock().unwrap().copy_and_paste))
```